presto-docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -0,0 +1,67 @@
+============================
+Lambdas and Lambda Functions
+============================
+
+Lambda Expression
+-----------------
+
+Lambda expressions in Presto are written with ``->``. For example::
+
+    x -> x + 1
+    (x, y) -> x + y
+    x -> regexp_like(x, 'a+')
+    x -> x[1] / x[2]
+    x -> IF(x > 0, x, -x)
+    x -> COALESCE(x, 0)
+    x -> CAST(x AS JSON)
+
+Most expressions you can write in Presto SQL can appear in lambda body, with a few exceptions:
+
+* ``TRY`` function is not supported yet. (:func:`TRY_CAST` is supported.)
+* Capture is not supported yet. In other words,
+
+  * Columns or relations can not be referenced.
+  * Only lambda variables from the inner-most lambda expression can be referenced.
+
+* Subqueries are not supported.
+
+Lambda Functions
+----------------
+
+.. function:: transform(array ARRAY(T), function FUNCTION(T,U)) -> ARRAY(U)
+
+    Returns an array that applies ``function`` to each element of ``array``. For example::
+
+        transform(ARRAY [], x -> x + 1) -> ARRAY []
+        transform(ARRAY [5, 6], x -> x + 1) -> ARRAY [6, 7]
+        transform(ARRAY [5, NULL, 6], x -> COALESCE(x, 0) + 1) -> ARRAY [6, 1, 7]
+        transform(ARRAY ['x', 'abc', 'z'], x -> CONCAT(x, '0')) -> ARRAY ['x0', 'abc0', 'z0']
+
+.. function:: filter(array ARRAY(T), function FUNCTION(T,BOOLEAN)) -> ARRAY(T)
+
+    Constructs an array from those elements of ``array`` for which ``function`` returns true. For example::
+
+        filter(ARRAY [], x -> true) -> ARRAY []
+        filter(ARRAY [5, -6, NULL, 7], x -> x > 0) -> ARRAY [5, 7]
+        filter(ARRAY [5, NULL, 7, NULL], x -> x IS NOT NULL) -> ARRAY [5, 7]
+        transform(ARRAY [ARRAY [1, NULL, 2], ARRAY[3, NULL]], a -> filter(a, x -> x IS NOT NULL)) -> ARRAY [ARRAY [1, 2], ARRAY [3]]
+
+.. function:: reduce(array ARRAY(T), initialState S, inputFunction FUNCTION(T,S,S), outputFunction FUNCTION(S,R)) -> R
```

Comment:
I agree having types is useful, but I find this hard to read. Let's see if we can improve this. Maybe breaking out the types into the description?

<img width="825" alt="screen shot 2016-12-07 at 10 04 43 am" src="https://cloud.githubusercontent.com/assets/9230/20980256/ade73d9c-bc64-11e6-843f-43d88b10ecdd.png">